rfcs/0000-kuadrant_crd.md

```diff
@@ -0,0 +1,164 @@
+policy support requires no policies to be present.
+
+If any of the following steps fails, all previously applied changes need to be unrolled and the
+`status` block of the CR would clearly reflected what changed and that Kuadrant is currently _not_
+operational. This is to be considered an atomic operation, either all gets successfully deployed or
+nothing is.
+
+If a policy is disabled but a policy is being added, when disallowing the admission, the webhook would
+use a response code `501 Not Implemented` with the additional `message` indicating the policy is
+disabled and as such can't be honored.
+
+### In an OSSM cluster
+
+The deployment of `Kuadrant` should fail if `authentication` isn't explicitly disabled.
+
+## State transitions of policy support
+
+If the `Kuadrant` object gets updated to change the support of a policy, the change would only be
+considered if no such policy exists in the cluster. No state transitions with regards to policy
+support is possible if policies of the given type are present.
+
+While it would be good to be able to enable the support for a policy while such policies already
+exist, there is much to consider as to what this would entail - here are a few things to consider:
+
+- can we apply all these policies?
+- what ordering should be considered to apply them?
+- what would a conflict mean?
+- what would disabling `authentication` for a cluster, while existing routes are currently protected by these?
+
+Requiring to first remove all policies and then only safely disable or enable the support for that
+kind of policy seems reasonable at this stage and can always be added later, which clearly specified
+behavior.
+
+### Disabling the support for a policy
+
+When disabling the support for a policy, merely the admission webhook's behavior changes. The
+`Authorino` resource or `Limitador` ones remain untouched.
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+The flipside of the flexibility this provides users with, this introduces quite some complexity. Hence
+the desire to keep this very rigid as to what are the preconditions to change the support for a
+policy.
+
+# Rationale and alternatives
+[rationale-and-alternatives]: #rationale-and-alternatives
+
+Possibly the gain of not running unnecessary services on the cluster isn't big enough for us to
+justify adding this complexity. Yet, as of today, we need a way to disable `authentication` in the
+case of running within an OSSM backed cluster.
+
+We could disable `authentication` on a heuristic base approach: running within OSSM? Disable
+`authentication`. The user experience seems less than desirable and confusing as best, as in future
+version this would probably change. That being said we're still pre-v1. We'd also need to find a way
+to know very early on what environment we are being deployed in.
+
+# Prior art
+[prior-art]: #prior-art
+
+While I couldn't find, probably because of lacking knowledge about similar use-cases, anything that
```

Comment:
I think in general not enabling an API in a cluster or disabling an API is done by making the CRD available or deleting it